rpms/SPECS/4.0.0/wazuh-manager-4.0.0.spec

```diff
@@ -0,0 +1,1108 @@
+fi
+
+exit 0
+%pre
+
+# Stop Authd if it is running
+if ps aux | grep %{_localstatedir}/ossec/bin/ossec-authd | grep -v grep > /dev/null 2>&1; then
+   kill `ps -ef | grep '%{_localstatedir}/ossec/bin/ossec-authd' | grep -v grep | awk '{print $2}'` > /dev/null 2>&1
+fi
+
+# Ensure that the wazuh-manager is stopped
+if [ -d %{_localstatedir}/ossec ] && [ -f %{_localstatedir}/ossec/bin/ossec-control ] ; then
+  %{_localstatedir}/ossec/bin/ossec-control stop > /dev/null 2>&1
+fi
+
+# Create the ossec group if it doesn't exists
+if command -v getent > /dev/null 2>&1 && ! getent group ossec > /dev/null 2>&1; then
+  groupadd -r ossec
+elif ! id -g ossec > /dev/null 2>&1; then
+  groupadd -r ossec
+fi
+# Create the ossec user if it doesn't exists
+if ! id -u ossec > /dev/null 2>&1; then
+  useradd -g ossec -G ossec -d %{_localstatedir}/ossec -r -s /sbin/nologin ossec
+fi
+# Create the ossecr user if it doesn't exists
+if ! id -u ossecr > /dev/null 2>&1; then
+  useradd -g ossec -G ossec -d %{_localstatedir}/ossec -r -s /sbin/nologin ossecr
+fi
+# Create the ossecm user if it doesn't exists
+if ! id -u ossecm > /dev/null 2>&1; then
+  useradd -g ossec -G ossec -d %{_localstatedir}/ossec -r -s /sbin/nologin ossecm
+fi
+
+if [ -d ${DIR}/var/db/agents ]; then
+  rm -f ${DIR}/var/db/agents/*
+fi
+
+# Remove existing SQLite databases
+rm -f %{_localstatedir}/ossec/var/db/global.db* || true
+rm -f %{_localstatedir}/ossec/var/db/cluster.db* || true
+rm -f %{_localstatedir}/ossec/var/db/.profile.db* || true
+rm -f %{_localstatedir}/ossec/var/db/agents/* || true
+
+# Remove Vuln-detector database
+rm -f %{_localstatedir}/ossec/queue/vulnerabilities/cve.db || true
+
+
+# Remove existing SQLite databases for Wazuh DB when upgrading
+# Wazuh only if upgrading from 3.2..3.6
+if [ $1 = 2 ]; then
+
+  # Import the variables from ossec-init.conf file
+  if [ -f %{_sysconfdir}/ossec-init.conf ]; then
+    . %{_sysconfdir}/ossec-init.conf
+  fi
+
+  # Get the major and minor version
+  MAJOR=$(echo $VERSION | cut -dv -f2 | cut -d. -f1)
+  MINOR=$(echo $VERSION | cut -d. -f2)
+
+  if [ $MAJOR = 3 ] && [ $MINOR -lt 7 ]; then
+    rm -f %{_localstatedir}/ossec/queue/db/*.db*
+    rm -f %{_localstatedir}/ossec/queue/db/.template.db
+  elif [ $MAJOR = 3 ] && [ $MINOR = 9 ]; then
+    find %{_localstatedir}/ossec/ruleset/sca -type f > %{_localstatedir}/ossec/old_sca.files
+  fi
+
+  # Delete old API backups
+  if [ -e %{_localstatedir}/ossec/~api ]; then
+    rm -rf %{_localstatedir}/ossec/~api
+  fi
+fi
+
+# Wazuh API is installed
+if [ -e %{_localstatedir}/ossec/api ]; then
+  # Stop API's services
+  if [ -n "$(ps -e | egrep ^\ *1\ .*systemd$)" ]; then
+    systemctl stop wazuh-api.service > /dev/null
+  elif [ -x /etc/rc.d/init.d/wazuh-api ] ; then
+    /etc/rc.d/init.d/wazuh-api stop > /dev/null
+  elif [ -n "$(ps -e | egrep ^\ *1\ .*init$)" ]; then
+    /etc/init.d/wazuh-api stop > /dev/null
+  fi
+  # Delete old Wazuh API service
+  rm /etc/init.d/wazuh-api
+fi
```

Comment:
There's a problem with this block. This block will be triggered always while upgrading a wazuh-manager-4.x package.

First, I would move this block to the beginning of the `%pre` and then, I would stop the `wazuh-manager` and `wazuh-api` services. Then, at the end of the `%pre` script, I would check if it is necessary to delete the `wazuh-api` service file, because this file should only be removed if there's a Wazuh API 3.x installed.